inbm/dispatcher-agent/dispatcher/sota/os_factory.py

```diff
@@ -137,10 +139,12 @@ def create_os_updater(self) -> OsUpdater:
         logger.debug("")
         return YoctoX86_64Updater()
 
-    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str], proceed_without_rollback: bool) -> Snapshot:
+    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str],
+                           proceed_without_rollback: bool, reboot_device: bool) -> Snapshot:
         logger.debug("")
         trtl = Trtl(PseudoShellRunner(), BTRFS)
-        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num, proceed_without_rollback)
+        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num,
+                             proceed_without_rollback, reboot_device)
```

Comment:
**Suggestion:** Refactor the repeated code in the create_snapshotter method implementations to a separate method to adhere to the DRY principle. [maintainability]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def create_snapshotter(self, sota_cmd: str, snap_num: Optional[str],
                           proceed_without_rollback: bool, reboot_device: bool) -> Snapshot:
        logger.debug("")
        return self._create_snapshotter(sota_cmd, snap_num, proceed_without_rollback, reboot_device)

    def _create_snapshotter(self, sota_cmd: str, snap_num: Optional[str],
                            proceed_without_rollback: bool, reboot_device: bool) -> Snapshot:
        trtl = Trtl(PseudoShellRunner(), BTRFS)
        return YoctoSnapshot(trtl, sota_cmd, self._dispatcher_broker, snap_num,
                             proceed_without_rollback, reboot_device)
```